pkgs/tools/networking/curl-impersonate/default.nix

```diff
@@ -1,27 +1,190 @@
+      })
+    ];
+
+    strictDeps = true;
+
+    nativeBuildInputs = [
+      installShellFiles
+      cmake
+      python3
+      python3.pkgs.gyp
+      ninja
+      perl
+      autoconf
+      automake
+      libtool
+      unzip
+      go
+      findutils
+      gnugrep
+      patchelf
+    ];
+
+    buildInputs = [
+      zlib
+    ];
+
+    configureFlags = [
+      "--with-ca-bundle=/etc/ssl/certs/ca-certificates.crt"
+      "--with-ca-path=${cacert}/etc/ssl/certs"
+      "--with-libnssckbi=$out/lib"
+    ];
+
+    buildFlags = [ "${target}-build" ];
+    checkTarget = "${target}-checkbuild";
+    installTargets = [ "${target}-install" ];
+
+    doCheck = true;
+
+    dontUseCmakeConfigure = true;
+    dontUseNinjaBuild = true;
+    dontUseNinjaInstall = true;
+    dontUseNinjaCheck = true;
+
+    postUnpack = lib.concatStringsSep "\n" (lib.mapAttrsToList (name: dep: "ln -sT ${dep.outPath} source/${name}") (lib.filterAttrs (n: v: v ? outPath) passthru.deps));
+
+    preConfigure = ''
+      export GOCACHE=$TMPDIR/go-cache
+      export GOPATH="$TMPDIR/go"
+      export GOPROXY=file://${passthru.boringssl-go-modules}
+      export GOSUMDB=off
+    '';
+
+    postInstall = ''
+      # Patch all shebangs of installed scripts
+      patchShebangs $out/bin
+
+      # Build and install completions for each curl binary
+      for bin in $out/bin/curl-impersonate-*; do
+        if [[ "$bin" =~ '-config' ]]; then
+          continue
+        fi
+
+        binname="$(basename "$bin")"
+
+        # Patch in correct binary name and alias it to all scripts
+        perl curl-*/scripts/completion.pl --curl $bin --shell zsh >"$TMPDIR/$binname".zsh
+        substituteInPlace "$TMPDIR/$binname".zsh \
+          --replace \
+            '#compdef curl' \
+            "#compdef $binname$(find $out/bin -name 'curl_*' -printf " %f=$binname")"
+
+        perl curl-*/scripts/completion.pl --curl $bin --shell fish >"$TMPDIR/$binname".fish
+        substituteInPlace "$TMPDIR/$binname".fish \
+          --replace \
+            '--command curl' \
+            "--command $binname$(find $out/bin -name 'curl_*' -printf " --command %f")"
+
+        # Install zsh and fish completions
+        installShellCompletion "$TMPDIR/$binname".{zsh,fish}
+      done
+    '';
+
+    preFixup = ''
+      # If libnssckbi.so is needed, link libnssckbi.so without needing nss in closure
+      if grep -F nssckbi $out/lib/libcurl-impersonate-*${stdenv.hostPlatform.extensions.sharedLibrary}; then
+        ln -s ${p11-kit}/lib/pkcs11/p11-kit-trust${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libnssckbi${stdenv.hostPlatform.extensions.sharedLibrary}
+        ${lib.optionalString (stdenv.hostPlatform.parsed.kernel.execFormat == lib.systems.parse.execFormats.elf)
+          "patchelf --add-needed libnssckbi${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libcurl-impersonate-*${stdenv.hostPlatform.extensions.sharedLibrary}"}
```

Comment:
I would move stdenv.hostPlatform.extensions.sharedLibrary to an alias in a let in to make the string block a bit shorter and cleaner